### ISSUE TITLE
Support  attention_mask  in multiHeadCrossAttentionPlugin

### ISSUE THREAD

**Issue by vivienfanghuagood** [NVIDIA/TensorRT#2587]
Hi，

I wonder if there are plans to support  _**attention_mask**_ in multiHeadCrossAttentionPlugin?

For pytorch users, the atten_mask or key_padding_mask is really useful in some scenes（like NLP generation tasks). So will Nvidia support these interfaces?
https://pytorch.org/docs/stable/generated/torch.nn.MultiheadAttention.html

**Comment by zerollzeng** [comment]
@ttyio ^ ^

**Comment by ttyio** [comment]
@vivienfanghuagood , we have `CustomQKVToContextPluginDynamic` for BERT like NLP task, and we expose `cuseqlens` input for the mask purpose https://github.com/NVIDIA/TensorRT/blob/main/plugin/bertQKVToContextPlugin/qkvToContextPlugin.cpp#L786

This cross attention is mainly for the Stable Diffusion, so we hide the `cuseqlens` to make the interface more simple, you need modify the code to expose the `cuseqlens` as mask https://github.com/NVIDIA/TensorRT/blob/main/plugin/multiHeadCrossAttentionPlugin/fmhcaPlugin.cpp#L99

**Comment by ttyio** [comment]
Closing since no activity for more than 3 weeks, please reopen if you still have question, thanks!
